### PR TITLE
[stdlib_math] Minor update to `stdlib_math` module and document

### PR DESCRIPTION
**Pull request by zoziha**
- [x] Minor changes to the internal implementation of `argd/argpi`;
- [x] Move  unit tests of `arange` procedure to `test_stdlib_math`;
- [x] Small update to the documentation.

#### Description

I found that there is some documentation in `stdlib_math` that needs to be updated in this PR.
Mainly related to:
- tests of `arange` function;
- `is_close/all_close` function FORD Links;
- The internal implementation of `argd/argpi` function, and `arg**` function FORD Links;
- `diff` function document.

Note: Since this is a somewhat minor modification, I believe it will be reviewed soon.

### CODE REVIEW

File: src/stdlib_math.fypp [hunk 1]
```diff
@@ -409,17 +416,18 @@ contains
         ${t1}$, intent(in) :: z
         real(${k1}$) :: result
 
-        result = merge(0.0_${k1}$, atan2(z%im, z%re), z == (0.0_${k1}$, 0.0_${k1}$)) &
-                 *180.0_${k1}$/PI_${k1}$
+        result = merge(0.0_${k1}$, atan2(z%im, z%re)*180.0_${k1}$/PI_${k1}$, &
+                 z == (0.0_${k1}$, 0.0_${k1}$))
```
Comment (zoziha) [inline src/stdlib_math.fypp:421]: `argd`: Putting the multiplication into the `merge` can reduce the unnecessary computation when `z=(0,0)`

File: src/stdlib_math.fypp [hunk 2]
```diff
@@ -409,17 +416,18 @@ contains
...
     end function argd_${k1}$
 
     elemental function argpi_${k1}$(z) result(result) 
         ${t1}$, intent(in) :: z
         real(${k1}$) :: result
 
-        result = merge(0.0_${k1}$, atan2(z%im, z%re), z == (0.0_${k1}$, 0.0_${k1}$)) &
-                 /PI_${k1}$
+        result = merge(0.0_${k1}$, atan2(z%im, z%re)/PI_${k1}$, &
+                 z == (0.0_${k1}$, 0.0_${k1}$))
```
Comment (zoziha) [inline src/stdlib_math.fypp:429]: `argpi`: same as `argd`

File: doc/specs/stdlib_math.md [hunk 3]
```diff
@@ -643,29 +643,26 @@ program demo_math_all_close
...
-```
-and for a rank-2 array
-```fortran
-y = [[stdlib_math(module):diff(interface)]](x [, n, dim, prepend, append])
-```
+For a rank-1 array:  
+`y = [[stdlib_math(module):diff(interface)]](x [, n, prepend, append])`
+
+and for a rank-2 array:  
+`y = [[stdlib_math(module):diff(interface)]](x [, n, dim, prepend, append])`
```
Comment (zoziha) [inline doc/specs/stdlib_math.md:665]: Make FORD link valid here.
\```fortran ... ``` -> \`...`

File: doc/specs/stdlib_math.md [hunk 4]
```diff
@@ -696,16 +693,17 @@ Shall be a `real/integer` and `rank-1/rank-2` array.
...
+- If the value of `n` is less than or equal to `0` (which is not recommended), the return value of `diff` is `x`.  
 - If the value of `dim` is not equal to `1` or `2` (which is not recommended),
 `1` will be used by the internal process of `diff`.
 
 
 #### Result value
 
 Returns the finite difference of the input array.
-Shall be a `real/integer` and `rank-1/rank-2` array.
+Shall be a `real/integer` and `rank-1/rank-2` array.
```
Comment (milancurcic) [inline doc/specs/stdlib_math.md:706]: Was this edit intentional? It shouldn't matter in Markdown.

**Review by milancurcic — APPROVED** [review]
Looks good, thanks. A small PR with no functional change.

File: doc/specs/stdlib_math.md [hunk 5]
```diff
@@ -320,7 +320,7 @@ program demo_logspace_rstart_cbase
 
 end program demo_logspace_rstart_cbase
 ```
-### `arange`
+### `arange` - Creates fixed-spaced values of given spacing, within a given interval
```
Comment (awvwgk) [inline doc/specs/stdlib_math.md:323]: Adding a description here breakes references to this function from the docs. Maybe just as separate line afterwards to keep the URL short?

Comment (zoziha) [reply doc/specs/stdlib_math.md:323]: I'm not sure if such a description needs to be created for every procedure, but it does make the URL longer.

Or I just describe: "`arange` function"? Just like "`clip` function".

**Comment by zoziha** [comment]
According to the review comments, relevant content has been revised.

It seems that the failure of FORD CI has nothing to do with this commit.
In my local version of FORD 6.15, there is no FORD failure.
The FORD version of CI is 6.17, I don't know if it is a FORD version problem.

**Review by jvdp1 — APPROVED** [review]
LGTM. Thank you.

**Comment by jvdp1** [comment]
Could you check why the actions failed please?

**Comment by zoziha** [comment]
The `ford` version was updated from 6.17 to 6.18, the old bugs were patched, and the new ones appeared:
- `exclude_dir: src/tests`, `*.fypp` inside `src/tests` should be ignored, but `ford` of this version 6.18 is trying to parse.

I'm wondering if `stdlib` should be selected to install the version number of `ford`, preferring a stable version number, such as 6.15.

It seems that the two `ford` documentation errors are related to `ford`'s version update, but I don't understand python code.

```sh
Reading file src/tests/linalg/test_linalg_matrix_property_checks.fypp
Preprocessing /Users/runner/work/stdlib/stdlib/src/tests/linalg/test_linalg_matrix_property_checks.fypp
Warning: error preprocessing /Users/runner/work/stdlib/stdlib/src/tests/linalg/test_linalg_matrix_property_checks.fypp
/Users/runner/work/stdlib/stdlib/src/tests/linalg/test_linalg_matrix_property_checks.fypp:1: error: include file 'common.fypp' not found [FyppFatalError]

call check(error, is_square(A_true), "0"): Found procedure call in MODULE
call check(error, (.not. is_square(A_false)), "0"): Found procedure call in MODULE
type(error_type), allocatable, intent(out) :: error: Found variable in SOURCEFILE
call check(error, is_diagonal(A_true_s), "0"): Found procedure call in SOURCEFILE
call check(error, (.not. is_diagonal(A_false_s)), "0"): Found procedure call in SOURCEFILE
call check(error, is_diagonal(A_true_sf), "0"): Found procedure call in SOURCEFILE
call check(error, (.not. is_diagonal(A_false_sf)), "0"): Found procedure call in SOURCEFILE
call check(error, is_diagonal(A_true_ts), "0"): Found procedure call in SOURCEFILE
call check(error, (.not. is_diagonal(A_false_ts)), "0"): Found procedure call in SOURCEFILE
end subroutine test_is_diagonal_${s1}$: END statement outside of any nesting
    main(proj_data, proj_docs, md)
  File "/usr/local/Cellar/ford/6.1.8/libexec/lib/python3.9/site-packages/ford/__init__.py", line 557, in main
    project = ford.fortran_project.Project(proj_data)
  File "/usr/local/Cellar/ford/6.1.8/libexec/lib/python3.9/site-packages/ford/fortran_project.py", line 104, in __init__
    ford.sourceform.FortranSourceFile(
  File "/usr/local/Cellar/ford/6.1.8/libexec/lib/python3.9/site-packages/ford/sourceform.py", line 1427, in __init__
    FortranContainer.__init__(self, source, "")
  File "/usr/local/Cellar/ford/6.1.8/libexec/lib/python3.9/site-packages/ford/sourceform.py", line 763, in __init__
    self._cleanup()
  File "/usr/local/Cellar/ford/6.1.8/libexec/lib/python3.9/site-packages/ford/sourceform.py", line 1069, in _cleanup
    raise NotImplementedError()
NotImplementedError
```

**Comment by zoziha** [comment]
It appears that because `*.fypp` in `src/tests` is parsed, using `include` explicitly currently works around this error temporarily.

Possibly related content:
- https://github.com/Fortran-FOSS-Programmers/ford/blob/5cc3abccaac54cedab2b42ae393a346aad48d474/CHANGELOG.md?plain=1#L149
- https://github.com/Fortran-FOSS-Programmers/ford/issues/259

**Comment by jvdp1** [comment]
Thank you @zoziha for debugging this FORD issue. This workaround seems to ok to me.
@awvwgk do you have additional comments?

**Review by awvwgk — APPROVED** [review]
